joblib/numpy_pickle_utils.py

```diff
@@ -176,57 +189,57 @@ def _read_fileobject(fileobj, filename, mmap_mode=None):
                       DeprecationWarning, stacklevel=2)
         yield filename
     else:
-        # Checking if incompatible load parameters with the type of file:
-        # mmap_mode cannot be used with compressed file or in memory buffers
-        # such as io.BytesIO.
-        if ((compressor in _COMPRESSORS or
-                isinstance(fileobj, tuple(_COMPRESSOR_CLASSES))) and
-                mmap_mode is not None):
-            warnings.warn('File "%(filename)s" is compressed using '
-                          '"%(compressor)s" which is not compatible with '
-                          'mmap_mode "%(mmap_mode)s" flag passed. mmap_mode '
-                          'option will be ignored.'
-                          % locals(), stacklevel=2)
-        if isinstance(fileobj, io.BytesIO) and mmap_mode is not None:
-            warnings.warn('In memory persistence is not compatible with '
-                          'mmap_mode "%(mmap_mode)s" flag passed. mmap_mode '
-                          'option will be ignored.'
-                          % locals(), stacklevel=2)
-
         # if the passed fileobj is in the supported list of decompressor
         # objects (GzipFile, BZ2File, LzmaFile), we simply return it.
         if isinstance(fileobj, tuple(_COMPRESSOR_CLASSES)):
-            yield fileobj
+            pass
```

Comment:
Actually I think we should remove this case now. This would allow for nested compression no?